src/service-object.ts

```diff
@@ -44,8 +43,11 @@ export interface Interceptor {
 
 // tslint:disable-next-line:no-any
 export type Metadata = any;
-export type MetadataResponse = [Metadata, r.Response];
-export type MetadataCallback =
+export type GetMetadataResponse = [Metadata, r.Response];
```

Comment:
So just to confirm, `RequestResponse` is `[r.Response, Metadata]` but `GetMetadataResponse` is `[Metadata, r.Response]`? :)

Reply:
Well, it actually makes sense after some thinking, since metadata is secondary for `RequestResponse` and is primary for `GetMetadataResponse`, but still... :)